hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/blockmanagement/BlockManager.java

```diff
@@ -3295,8 +3295,11 @@ private BlockInfo processReportedBlock(
     BlockInfo storedBlock = getStoredBlock(block);
     if(storedBlock == null) {
       // If blocksMap does not contain reported block id,
-      // the replica should be removed from the data-node.
-      toInvalidate.add(new Block(block));
+      // the replica should be removed from the data-node,
```

Comment:
The replica should be removed from Datanode, and set NumBytes to BlockCommand.No_ACK to avoid useless report to NameNode from Datanode when complete to process it.

Reply:
Sorry for the late reply.
Noted, I will  update PR today, Thanks.

Reply:
Sir,  please help reivew it again, Thanks.